Sources/Datadog/Tracing/DDSpan.swift

```diff
@@ -84,33 +68,35 @@ internal class DDSpan: OTSpan {
     }
 
     func setOperationName(_ operationName: String) {
-        if warnIfFinished("setOperationName(_:)") {
-            return
+        queue.async {
```

Comment:
I wonder why do you need to run these operations in an Async Queue anyway ? Is it because it is an expensive operation under the hood ?

Reply:
It's because we access the shared state: `unsafeOperationName` property. As this is a public API and can be called from any thread, we need to synchronize it. When it comes to running operation on `queue`, there are two choices: `queue.sync {}` or `queue.async {}`. The prior is blocking the caller thread and the latter does not, so `async` will perform better for the user and do what needs to be done for us.

Speaking Android, I think this is use of `queue` is quite similar to Java `@synchronised` annotation - we use it as a locking mechanism. Although there are [few other choices available for iOS](https://dmytro-anokhin.medium.com/concurrency-in-swift-reader-writer-lock-4f255ae73422) (including raw POSIX locks), `queues` are well optimised and can be equally performant as other locking mechanisms.